src/components/navigation-bar/navigation-bar.test.js

```diff
@@ -0,0 +1,169 @@
+  return <NavigationBar {...props}>{isChildren}</NavigationBar>;
+};
+
+context("Testing NavigationBar component", () => {
+  describe("should render NavigationBar component", () => {
+    it.each(specialCharacters)(
+      "should render NavigationBar component with %s as a children",
+      (childrenValue) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent>{childrenValue}</NavigationBarComponent>
+        );
+
+        navigationBar().should("have.text", childrenValue);
+      }
+    );
+
+    it("should render NavigationBar component with ariaLabel prop", () => {
+      CypressMountWithProviders(
+        <NavigationBarComponent ariaLabel={testData} />
+      );
+
+      navigationBar().should("have.attr", "aria-label", testData);
+    });
+
+    it.each(variants)(
+      "should render NavigationBar component with %s as a navigationType",
+      (navigationType, color) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent navigationType={navigationType} />
+        );
+
+        navigationBar().should("have.css", "background-color", color);
+      }
+    );
+
+    it.each([
+      [true, "not.exist"],
+      [false, "be.visible"],
+    ])(
+      "should render NavigationBar component with isLoading prop set to %s",
+      (boolean, assertion) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent isLoading={boolean}>
+            {testData}
+          </NavigationBarComponent>
+        );
+
+        cy.contains(testData).should(assertion);
+      }
+    );
+
+    it.each([["fixed"], ["sticky"]])(
+      "should render NavigationBar component with position prop set to %s with orientation set to top",
+      (position) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent position={position} orientation="top" />
+        );
+
+        navigationBar().should("have.css", "position", position);
+      }
+    );
+
+    it.each([["fixed"], ["sticky"]])(
+      "should render NavigationBar component with position prop set to %s with orientation set to bottom",
+      (position) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent position={position} orientation="bottom" />
+        );
+
+        navigationBar().should("have.css", "position", position);
+      }
+    );
+
+    it.each([[25], [100], [-100]])(
+      "should render NavigationBar component with offset prop set to %s",
+      (offset) => {
+        CypressMountWithProviders(<NavigationBarComponent offset={offset} />);
+
+        navigationBar().should("have.attr", "offset", offset);
+      }
+    );
+
+    it.each([["top"], ["bottom"]])(
+      "should render NavigationBar component with orientation prop set to %s",
+      (orientation) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent orientation={orientation} />
+        );
+
+        navigationBar().should("have.attr", "orientation", orientation);
```

Comment:
as in the above comment, I think it's just an accident that the rendered HTML element has an `orientation` attribute with this value. Orientation is actually just used to decide whether to use CSS `top` or `bottom` for the `offset` value.